crawler/db.py

```diff
@@ -189,16 +191,12 @@ def create_mysql_connection(config: ModuleType, readonly=True) -> CMySQLConnecti
             # default is false, but specify it so more predictable
             use_pure=False,
         )
-        if mysql_conn is not None:
-            if mysql_conn.is_connected():
-                logger.debug("MySQL Connection Successful")
-            else:
-                logger.error("MySQL Connection Failed")
+        logger.debug("MySQL Connection Successful")
 
+        return cast(CMySQLConnection, mysql_conn)
     except mysql.Error as e:
-        logger.error(f"Exception on connecting to MySQL database: {e}")
-
-    return mysql_conn
+        logger.error(f"Exception while connecting to MySQL database: {e}")
+        raise
```

Comment:
Not sure you should change the behaviour here without changing the error handling of the code that calls it - have you followed it up the chain?

Reply:
Good spot thanks, will dig a little deeper.

Reply:
Reverted to larger refactor and from quick inspection does not seem like the correct behaviour to raise here.